itests/mir_test.go

```diff
@@ -49,10 +50,196 @@ func setupMangler(t *testing.T) {
 	})
 }
 
-// func TestMirMessageFromLearner(t *testing.T) {
-// 	ts := itestsConsensusSuite{opts: []interface{}{kit.ThroughRPC(), kit.MirConsensus()}}
-// 	t.Run("testMirMessageFromLearner", ts.testMirMessageFromLearner)
-// }
+// TestMirWithOneNodeReconfiguration tests that the reconfiguration mechanism operates normally
+// if a new validator joins the network and then leaves it.
+func TestMirWithOneNodeReconfiguration(t *testing.T) {
+	var wg sync.WaitGroup
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer func() {
+		t.Logf("[*] defer: cancelling %s context", t.Name())
+		cancel()
+		wg.Wait()
+	}()
+
+	membershipFileName := "_membership_config.tmp"
```

Comment:
I see you are using this code in several places, what about making it its own function? If we figure out a better way of doing this in the future we will have to change it all over the place.

Reply:
Fixed